File: School Classes/Programming 2(Java)/Module_6/src/lecture2/Vehicle.java
//abstract class that represents concept of vehicle

package lecture2;

public abstract class Vehicle {
	//class methods
	public void start() {
		System.out.println("Vehicle Started.");
	}
	
	public void stop() {
		System.out.println("Vehicle Stopped.");
	}
	
	//abstract methods
	public abstract double fuelEfficiency();

}

File: School Classes/Python/Module 4/Lab 2/vehicle.py
# Lane Pollock
# 11 Mar 2024
# Lab 4.2 - Vehicle Inheritance Lab
# vehicle class to hold make model and year as well as general methods

class Vehicle:
    def __init__(self, make, model, year):
        self.make = make
        self.model = model
        self.year = year
    
    #method to start engine
    def start_engine(self):
        print("Engine started.")

    #stop engine
    def stop_engine(self):
        print("Engine stopped.")

    #print details
    def display_details(self):
        print(f"{self.year} {self.make} {self.model}")
